src/Http/Http.Abstractions/src/PathString.cs

```diff
@@ -172,8 +175,10 @@ private static string ToEscapedUriComponent(string value, int i)
         /// <returns>The resulting PathString</returns>
         public static PathString FromUriComponent(string uriComponent)
         {
-            // REVIEW: what is the exactly correct thing to do?
-            return new PathString(Uri.UnescapeDataString(uriComponent));
+            Span<char> pathBuffer = uriComponent.Length <= StackAllocThreshold ? stackalloc char[StackAllocThreshold] : new char[uriComponent.Length];
```

Comment:
Is this entire function non allocating if there's nothing to escape? I think that's an important property to preserve.

Reply:
I covered this case now.